README.md
# phpxmail-2

PhpXmail is a web based management software for the Xmail mail server. It provides a GUI (Graphic User Interface) for the Xmail administration extensions.

It allows the administrator of the mail server to perform configuration management and monitoring tasks. It allows the postmaster for each domain to perform management functions. It allows users to manage their account settings.

The original sources from Vlad Alexa have been ported to PHP7.

## Installation 

See [install.txt](https://github.com/morphoiz/phpxmail-2/blob/master/install.txt) file.

## Licensing

This program is published under the terms of the GNU General Public License v2.0.

## Credits

Based on [PHPXmail](https://sourceforge.net/projects/phpxmail/) written by Vlad Alexa Mancini.


changelog.txt
1.5
2007-06-09
fixed a bug: on some systems it needs to login twice... - (below_zero)
fixed a bug: solved a problem with IE7 - (dsion)
fixed a bug: "Strange behaviour of update link" it does not always correctly detect the "Updated version" - (filip1258, below_zero)
fixed a bug: Xmail report all mails as spam when PHPXmail empty 'IP spamlist' or 'Addresses spamlist'. - (makelt, below_zero)
improved code: in LOGIN.INC the use of a boolean variable was not formally correct. - (myslug, below_zero)
fixed a bug: phpxmail doesn't recognize user/mailinglist correctly if the account type is in lower case - (jstuckle, below_zero)
fixed capability to work also on that systems were in PHP.INI "short_open_tags" is "OFF" - (jstuckle, below_zero)
fixed a bug: an user was not able to register itself if in his domain settings MaxMailBoxes=0 (unlimited) - (below_zero)
improved regular User Login: who logs-in with a mailing list Id & Pwd now may ADD, EDIT & DELETE addresses - (csmorier, below_zero)
fixed a bug: an extra 'carriage return' were saved in mailproc.tab file  - (csmorier, below_zero)
added defaults for a correct environment setting (temporary they will overcome those in php.ini) - (below_zero)
fixed a bug: on certain systems 'filterlist' page gives error or warning - (pierre-oord, below_zero)
fixed a bug when edit, and disable, an user 'POP3 link' - (below_zero)

1.4
2005-10-16
improved servers info: now users are noticed about wrong settings in 'CTRL account' field  - (below_zero)
added a text file on how to install this package  - (below_zero)
improved mailing list management: user rights are editable only when mailing list is 'Closed', enhanced interface - (below_zero)
fixed a couple of bugs in the IP_Object class  - (below_zero)
improved SMTP log parsing: clicking on the Sender IP will automatically add it to the local spam IP list - (below_zero)
added control for Xmail configuration file SPAMMERS.TAB - (below_zero)
improved SMTP log parsing: clicking on the Sender Address will automatically add it to the local spam address list - (below_zero)
added control for Xmail configuration file SPAM-ADDRESS.TAB - (below_zero)
fixed a series of bugs when 'register_globals' in php.ini is 'off' - (uniboee, phpn, mancini & below_zero)
improved alias management: now the administrator can limits the max number of aliases for all users in a domain - (below_zero)
improved 'change settings' for users: now a regular user may view/add/delete aliases for his main mailbox - (below_zero)
improved login info: added a warning about a wrong setting of 'session.use_cookies' in PHP.INI - (pengu - below_zero)
fixed mailbox limiting: now a visitor cannot register itself to a domain that have reached the max mailboxes limit - (below_zero)
improved 'domain list': now quantity of users and mailing lists are splitted in two coloumns - (below_zero)
internal rearrangement: moved two new PHPXmail misc functions from the file 'class.xmail.php' to 'class.phpxm.php' - (below_zero)

1.3
2005-08-21
improved domain management: now the administrator can limits the max number of mailboxes for the postmaster - (below_zero)
improved login process: if it's needed, the user is noticed that he must enable cookies before login - (below_zero)
improved mailing list management: added a page for edit the vars of a mailinglist account - (below_zero)
improved mailing list management: added a page for edit the user rights of a mailinglist - (below_zero)
improved mailing list management: now is possible to set also the 'auth SMTP' in 'write' right of an user - (below_zero)
improved log parsing: now it's possible to empty the records into the log files - (below_zero)
improved 'mail processes' interfaces: now they use also the vertical bar | for real tabs - (below_zero)
improved 'filter file edit': now it use also the vertical bar | for real tabs (delimiting double quotes are stripped) - (below_zero)
improved 'file list' in 'server filters': an usage map for Pre/Post/In/Out filters is now shown - (below_zero)
fixed 'Filter panel control': after deleting all the filters in a Pre/Post/In/Out filter, 'Count' value was 1, now is 0 - (below_zero)
fixed security: the default value of 'Allow visitors to register a mail account' in 'phpxmail config' is now OFF - (below_zero)

1.2
2005-07-13
fixed a authentication bypass security flaw reported by Stefan Lochbihler (mancini)
fixed a bug when the login password is typed erroneously: PHPXmail hangs in a endless 'Denied Access' loop - (below_zero)
improved first start-up: the initial page is the 'Add server' page, Login is permitted only after a server is added - (below_zero)
fixed a bug when adding more than one user in a domain: the Nav-menu erroneously report him as a mailing list user - (below_zero)

1.1
2005-06-02
Fixed empty fields submitting when adding a server: now all fields must be filled - (below_zero)
added advanced log parsing of all the log files (selectable by date), error info & color highlighting - (below_zero)
fixed user menu bug (mailbox name was not shown in nav-menues) - (below_zero & phpn) 
fixed an 'Auth deny' in certain circumstances when using virtual nodes in nav. caption - (below_zero)
improved mail processes interface (usergetmproc.inc & domainmproc.inc) now it's not required to delimit fields with " - (below_zero) 
fixed a bug in usergetmproc.inc file (does not delete mailproc.tab when it is empty)so Xmail server loss e-mails - (below_zero)
improved graphics of 'Server Config Variables': uniformed to 'user defaults' (see below) - (below_zero)
improved graphics of 'User Defaults': Now variables are enabled toggling ON/OFF icons, disabled variables are grayed - (below_zero)
fixed a bug that does not permit to setup server parameters for those users that make a clean install of ver. 1.0 - (below_zero)

menu.php
<?php

require_once('class.xmail.php');
require_once('config.php');
require_once('environment.inc');
session_start();

if ($cfg_debug_stat == "on") {
	error_reporting(E_ALL);
}

if (isset($_SESSION['mail_server']))
$mail_server = $_SESSION['mail_server'];
else      $mail_server = new XMail();
?>

<html>
<head>
    <title></title>
    <link rel="stylesheet" href="main.css" type="text/css">
	<base target='main'>
</head>

<body class="menu">

<img src='gfx/phpx.gif' alt='phpxmail' border='0'>

<div class='content'>

<?php

echo "<div align=center><b>PHPXmail&nbsp;<a href=http://nextcode.org/forums/ target=_blank>support</a><br>";

echo "Version: $cfg_version_num";

if ($cfg_update_chk == "on"){
	$available = file ('http://phpxmail.sourceforge.net/version.txt');
	$installed = $cfg_version_num;
	if ($available){
		if ($installed < rtrim($available[0])) {
			echo "&nbsp<img src=gfx/ico_warn.gif align=absmiddle>&nbsp<a href='main.php?action=update'> update </a></b>";
		}
	}
} else {
	echo "&nbsp<a href='main.php?action=update'> update </a>";
}

echo "</div>\n";

if (isset($_SESSION['logged_in'])){
	if ($_SESSION['logged_in'] == "The operation completed successfully.") {

		if ($mail_server->noop()){
			echo "<div align=center><br><b>Xmail server: " .$mail_server->xm_ver ."</b></div><br><br>";
		}

		switch ($mail_server->xm_ctrl_t) {
			case 'r_user': // user menu
			echo "<a href='main.php?action=userchange'><div class='menulink'>change settings</div></a>";
			break;

			case 'd_admin': // domain admin menu
			echo "<a href='main.php?action=userlist'><div class='menulink'>users</div></a>";
			break;

			case 's_admin': // server admin menu
			echo "<a href='main.php?action=servervars'><div class='menulink'>server config</div></a>\n";
			echo "<a href='main.php?action=domainlist'><div class='menulink'>server domains</div></a>\n";
			echo "<a href='main.php?action=serverfilt'><div class='menulink'>server filters</div></a>\n";
			echo "<a href='main.php?action=spamlist'><div class='menulink'>server spam lists</div></a>\n";
			echo "<a href='main.php?action=serverlogs'><div class='menulink'>server logs</div></a>\n";
			echo "<a href='main.php?action=frozlist'><div class='menulink'>server spool</div></a>\n";
			echo "<br>\n";
			echo "<a href='main.php?action=config'><div class='menulink'>phpxmail config</div></a>\n";
			echo "<a href='main.php?action=serverlist'><div class='menulink'>phpxmail setup</div></a>\n";
			break;
		}

		echo "<br>\n";
		echo "<a href='main.php?action=logout'><div class='menulink'>logout</div></a><br>\n";

	}  else { // show login

	echo "<br>\n";
	echo "<a href='main.php?action=login'><div class='menulink'>login</div></a><br>\n";

	}
} else {
	echo "<br>\n";
	// show login (if was set almost one server)
	if (count(serverlist($cfg_servers))){
		echo "<a href='main.php?action=login'><div class='menulink'>login</div></a><br>\n";

		//show register
		if ($cfg_visitor_reg == "on"){
			echo "<a href='main.php?action=register'><div class='menulink'>register</div></a><br>";
		}
	}
}
if (count(serverlist($cfg_servers)) == 0){
	echo "<a href='main.php?action=serverlist'><div class='menulink'>setup</div></a><br>";
}

echo "<a href='main.php?action=legend'><div class='menulink'>legend</div></a><br>";

?>

</div>

</body>
</html>

checkssh.php
<?php
	
	require('class.xmail.php');
	session_start();
	$mail_server = $_SESSION['mail_server'];

	if ($cfg_debug_stat == "on") {
	error_reporting(E_ALL);
	}

	$user = $_REQUEST['user'];
	$server = $_REQUEST['server'];
	$ssh_success = test_ssh($user, $server);

?>

<html>
<head>
	<title>SSH check</title>
	<link rel="stylesheet" href="gfx/main.css" type="text/css">
	<script>window.resizeTo(400,150)</script>
</head>
<body>

<div class='content'>
<?php if ($ssh_success) { ?>
	Connection to <?php echo $server; ?> was a success<br>
<?php } else { ?>
	Could not make a connection to <?php echo $server; ?><br>
	<li>check if <?php echo $server; ?> is in known_host on <?php echo $_SERVER["SERVER_NAME"]; ?>
	<li>check if <?php echo $user; ?> is in authorized_keys <?php echo $server; ?><br>
<?php } ?>
<a href="#" onclick="window.close()">close window</a>
</div>
</body>
</html>

main.php
<?php
require_once('class.xmail.php');
require_once('config.php');
require_once('environment.inc');
session_start();

// in the future releases the following statement may be
// transferred in the 'CONFIG.PHP' and then managed
// by the page 'CONFIG.INC'
$cfg_allow_caption_info='on';

if ($cfg_debug_stat == "on") {
	error_reporting(E_ALL);
}

// import some vars
if (isset($_SESSION['mail_server']))
$mail_server = $_SESSION['mail_server'];
else      $mail_server = new XMail();

if (isset($_REQUEST['action'])){
	$action = $_REQUEST['action'];
} else {
	if (count(serverlist($cfg_servers))){
		$action = 'login';
	} else {
		$action = 'serverlist';
	}
}

if (isset($_REQUEST['domain']))
$domain = $_REQUEST['domain'];
else     $domain = '';

if (isset($_REQUEST['username']))
$username = $_REQUEST['username'];
else     $username = '';

// set defaults to prevent url spoofing
switch ($mail_server->xm_ctrl_t) {
	case 'r_user':
		$domain = $mail_server->xm_user_d;
		$user = $mail_server->xm_user_u;
		$type = 'r_user';
		if (empty($username)) $username = $user;
	case 'd_admin':
		$domain = $mail_server->xm_user_d;
		$type = 'd_admin';
	case 's_admin':
		$type = 's_admin';
}

//if (empty($_POST) && $action =="login") {
if ($action =="login") {
	// Set session cookie: later it will be
	// checked if session cookie has been blocked or not
	// (Be sure that on the server side in php.ini the "session.use_cookies" is set to true)
	if (empty($_GET['reloaded'])) {
		$_SESSION['InfoAreReadable'] = true;
		// Reload the page, so session values will be available
		header ("Location: main.php?action=login&reloaded=yes");
	}
}

// get/set some vars (usefull for the dynamic part of menu titles...)
if (isset($_REQUEST['ftype'])) $filtertype = $_REQUEST['ftype']; else $filtertype = '';
if (isset($_REQUEST['svr'])) $svr = $_REQUEST['svr']; else $svr = '';
if (isset($_REQUEST['alias'])) $alias = $_REQUEST['alias']; else $alias = '';
if (isset($_REQUEST['mlname'])) $mlname = $_REQUEST['mlname']; else $mlname = '';
if (isset($_REQUEST['log'])) {
	$ltype = $_REQUEST['ltype'];
	$log = $_REQUEST['log'];
} else {
	$ltype = '';
	$log = 'Empty Log';
}
if (isset($_GET['ltype'])){
	switch ($_GET['ltype']) {
		case 'ctrl': $logtype = 'Administration Server log records';	break;
		case 'fing': $logtype = 'Finger Server log records';	break;
		case 'filters': $logtype = 'SMTP Filter log records';	break;
		case 'lmail': $logtype = 'Local Mail Server log records';	break;
		case 'pop3': $logtype = 'Pop3 Server log records';	break;
		case 'psync': $logtype = 'Pop3 Synch Server log records';	break;
		case 'smail': $logtype = 'SendMail Server log records';	break;
		case 'smtp': $logtype = 'SMTP Server log records';	break;
		default: break;
	}
}

?>
<html>
<head>
        <link rel="stylesheet" href="main.css" type="text/css">
</head>
<body>
<img src='gfx/mail.gif' height='99' width='450' border='0'>
<div class='content'>

<?php


//#####################################################################################################
require_once('class.phpxm.php');
$mnu = new MenuTree();

// DEFINE USER TYPES AND THEIRS BINARY AUTH WEIGHTS (1,2,4,8,16,..,2^n)
$mnu->AddUserType('anonymous', 1);
$mnu->AddUserType('r_user', 2);
$mnu->AddUserType('d_admin', 4);
$mnu->AddUserType('s_admin', 8);

// DEFINE MENU STRUCTURE =============================================================================================================
// (best view with 'tab' of 4 characters)
// AUTH is in binary form and it represents the sum of the AUTH weigths (may be used also a decimal value)
// examples: '1000' => s_admin; '1100' => s_admin + d_admin; '0010' => r_user
// note: '1000' = 8; '0100' = 4; '1100' = 12 (8+4); '0010' = 2
//
//  ->Add('MenuAction'  'FileToInclude.inc'		 'AUTH'	 'ParentMenu'   #  'Title \t subtitle'  'parameters'  'Info tips/caption)';

// 000 - login files
$mnu->Add('login',		'login.inc', 			 '1111', '', 	   		0, 'Login', '', '');		// login to server
$mnu->Add('logout',		'logout.inc', 			 '1111', '', 	   		0, 'Logout', '', '');	// logout from server

// 100 - domain files
$mnu->Add('domainlist',	'domain/domainlist.inc', '1000', '', 	   		0, 'server domains|domains', '', 'List of all the served domains');	// list domains
$mnu->Add('domainadd',	'domain/domainadd.inc',  '1000', 'domainlist', 	0, 'add new domain', '', '');	// create new domain
$mnu->Add('domaindel',	'domain/domaindel.inc',  '1000', 'domainlist', 	0, 'remove "$domain"', '', '');	// delete domain
$mnu->Add('custdomdel',	'domain/domaindel.inc',  '1000', 'domainlist', 	0, 'remove custom domain', '', '');	// delete custom domain
$mnu->Add('custdomget',	'domain/custdomget.inc', '1000', 'domainlist', 	0, '$domain (custom domain)', '', '');	// get custom domain
$mnu->Add('custdomset',	'domain/custdomset.inc', '1000', 'domainlist', 	0, 'add new custom domain', '', '');	// set custom domain

// 200 - user files
$mnu->Add('userlist',	'user/userlist.inc', 	 '1100', 'domainlist', 	1, '$domain \t mailboxes', '&domain=$domain', 'List of all the mailboxes of this domain');	// list users
$mnu->Add('vxnode202',	'', 	 				 '1100', 'domainlist', 	2, '$domain \t mailing lists', '', '');	// start submenu mailing lists
$mnu->Add('domuserdef',	'usrdefaults.inc', 		 '1100', 'domainlist', 	3, '$domain \t users defaults', '&domain=$domain', 'Variables and values that must be used by default when creating new users in $domain');	// set default values for users
$mnu->Add('domainmproc','domain/domainmproc.inc','1100', 'domainlist', 	4, '$domain \t domain mail proc', '&domain=$domain', 'Mail processes that are common for all users in this domain');	// set default values for users
$mnu->Add('domsettings','domain/domsettings.inc','1000', 'domainlist', 	9, '$domain \t domain settings', '&domain=$domain', 'Special settings for the domain <b>"$domain"</b>');	// set default values for users
$mnu->Add('useradd',	'user/useradd.inc', 	 '1100', 'userlist',   	0, 'add new user', '&domain=$domain', '', '');	// add user / maillist
$mnu->Add('userdel',	'user/userdel.inc', 	 '1100', 'userlist',   	0, 'remove user', '&domain=$domain&username=$username', '', '');	// delete user / maillist

// 201 - user files MAILBOXES
$mnu->Add('uservars',	'user/uservars.inc', 	 '1100', 'userlist',   	1, '$username@$domain \t account', '&domain=$domain&username=$username', 'Account settings for <b>"$username@$domain"</b>');	// user vars
$mnu->Add('userchange',	'user/userchange.inc',   '0010', 'userlist',    1, '$username@$domain \t settings', '&domain=$domain&username=$username', 'Change settings for "$username@$domain"');	 // user allowed changes
$mnu->Add('aliaslist',	'user/aliaslist.inc', 	 '1110', 'userlist',   	2, '$username@$domain \t aliases',  '&domain=$domain&username=$username', 'Virtual names (aliases) settings for <b>"$username@$domain"</b>');	// list aliases
$mnu->Add('poplnklist',	'user/poplnklist.inc',   '1100', 'userlist',   	3, '$username@$domain \t ext POP3', '&domain=$domain&username=$username', 'Preset an external mailboxes from which to collect messages');	// list pop3s
$mnu->Add('pop3ipmap',	'user/pop3ipmap.inc',    '1100', 'userlist',   	4, '$username@$domain \t pop clients', '&domain=$domain&username=$username', 'IP settings from which a POP3 client may access to this mailbox');	// user vars
$mnu->Add('usergetmproc','user/usergetmproc.inc','1100', 'userlist',    5, '$username@$domain \t user mail proc', '&domain=$domain&username=$username', 'Mail processes settings for <b>"$username@$domain"</b> mailbox');	// user IP (POP3) permissions
$mnu->Add('aliasadd',	'user/aliasadd.inc', 	 '1110', 'aliaslist',  	0, 'add new alias',  '&domain=$domain&username=$username', '');	// add alias
$mnu->Add('aliasdel',	'user/aliasdel.inc', 	 '1110', 'aliaslist',  	0, 'remove alias $alias',  '&domain=$domain&username=$username&alias=$alias', '');	// remove alias
$mnu->Add('poplnkadd',	'user/poplnkadd.inc',    '1110', 'poplnklist', 	0, 'add new POP3', '&domain=$domain&username=$username', '', '');	// add pop3
$mnu->Add('poplnkdtl',	'user/poplnkdtl.inc',    '1110', 'poplnklist', 	0, '$p_username@$p_domain', '', '');	// pop3 dtl
$mnu->Add('poplnkdel',	'user/poplnkdel.inc',    '1110', 'poplnklist', 	0, 'remove POP3 link', '', ''); // del pop3 link

// 202 - user files MAILING LIST
$mnu->Add('maillists',	'user/maillists.inc', 	 '1100', 'vxnode202', 	1, 'mailing lists', '&domain=$domain', 'List of all the mailing lists of this domain');	// list mailing lists
$mnu->Add('mluservars',	'user/mluservars.inc',	 '1100', 'maillists', 	1, '$mlname \t account', '&domain=$domain&mlname=$mlname', 'Account for the mailing list');	// list mailing lists
$mnu->Add('mluserlist',	'user/mluserlist.inc',   '1100', 'maillists',   2, '$mlname \t addresses', '&domain=$domain&mlname=$mlname', 'List of mailing addresses'); // list mailing list users
$mnu->Add('mailtolist',	'sendmail.inc', 		 '1100', 'maillists', 	3, '$mlname \t send mail to list', '&domain=$domain&mlname=$mlname', 'Send a message to the whole mailing list'); // send mail
$mnu->Add('mllistdel',	'user/mllistdel.inc',    '1100', 'maillists', 	0, 'remove', '&domain=$domain&username=$username', ''); // del mailing list
$mnu->Add('mllistadd',	'user/useradd.inc', 	 '1100', 'maillists',  	0, 'add new list', '&domain=$domain', '', '');	// add maillist
$mnu->Add('mluseradd',	'user/mluseradd.inc',    '1100', 'mluserlist', 	0, 'add address', '', 'Add a new user address to the mailing list'); // add mailing list users
$mnu->Add('mluseredit',	'user/mluseredit.inc',   '1100', 'mluserlist', 	0, 'edit address', '', 'Edit user rights for this address'); // edit mailing list users
$mnu->Add('mluserdel',	'user/mluserdel.inc',    '1100', 'mluserlist', 	0, 'remove address', '', ''); // del mailing list users

// 300 - send mail
$mnu->Add('sendmail',	'sendmail.inc', 		 '1100', 'mluserlist', 	0, 'send message', '', ''); // send mail

// 400 - servers setup
$mnu->Add('serverlist',	'server/serverlist.inc', '1001', '', 	   		0, 'phpxmail setup|setup', '', 'List of Xmail servers to which PHPXmail it can be connected');	// list servers
$mnu->Add('serveradd',	'server/serveradd.inc',  '1001', 'serverlist', 	0, 'add server', '', 'Add an Xmail server to the PHPXmail connection list');	// add server
$mnu->Add('serverdel',	'server/serverdel.inc',  '1000', 'serverlist', 	0, 'remove server', '', 'Remove an Xmail server from the PHPXmail connection list');	// remove server
$mnu->Add('servercfg',	'server/servercfg.inc',  '1000', 'serverlist', 	0, '$svr', '', '');	// configure server

// 401 - server configs
$mnu->Add('vxnode401',	'', 					 '1000', '', 	   		0, '', '', '');	// push submenu down of 1 level (disconnect from root level)
$mnu->Add('servervars',	'server/servervars.inc', '1000', 'vxnode401',  	1, 'server config \t config', '', 'Xmail server configuration variables');	// list server config
$mnu->Add('usrdefaults','usrdefaults.inc', 		 '1000', 'vxnode401',  	4, 'server config \t users defaults', '', 'Global default variables and values that must be used when creating a new user<br>(They are common for all the served domains, unless specific domain settings)');	// set default values for users
$mnu->Add('vxnode402',	'', 	 				 '1000', 'vxnode401',  	3, 'server config \t IP maps', '', '');	// group IP-MAPs in a sub menu
$mnu->Add('servercmd',	'server/servercmd.inc',	 '1000', 'vxnode401',  	5, 'server config \t commands', '', 'Miscellaneous commands for the Xmail server');	// send commands to xmail server

// 402 - server IP Access Settings
$mnu->Add('srvipmpop3',	'server/ipmap.inc', 	 '1000', 'vxnode402',  	2, 'IP maps \t POP3', '&iptype=pop3', 'POP3 server global access settings');	// Global POP3 access
$mnu->Add('srvipmsmtp',	'server/ipmap.inc', 	 '1000', 'vxnode402',  	1, 'IP maps \t SMTP', '&iptype=smtp', 'SMTP server global access settings');	// Global SMTP access
$mnu->Add('srvipmfinger','server/ipmap.inc', 	 '1000', 'vxnode402',  	3, 'IP maps \t Finger', '&iptype=finger', 'FINGER server access settings'); //Global FINGER access
$mnu->Add('srvipmctrl', 'server/ipmap.inc', 	 '1000', 'vxnode402',  	4, 'IP maps \t Control', '&iptype=ctrl', 'Xmail remote control access settings'); //server CONTROL access

// 500 - server filters
$mnu->Add('vxnode501',	'', 					 '1000', '', 	   		0, '', '', '');	//  push submenu down of 1 level (disconnect from root level)
$mnu->Add('serverfilt',	'server/serverfilt.inc', '1000', 'vxnode501', 	1, 'server filters \t filters', '', 'Filters panel control');	// server filters list

$mnu->Add('serverfilters','server/serverfilters.inc','1000','serverfilt',0, '$filtertype', '', '');	// server filters
$mnu->Add('filterlist', 'server/filterlist.inc', '1000', 'vxnode501',   2, 'server filters \t files list', '', 'List and usage of all filter files available on the server');
$mnu->Add('filterdel',	'server/filterdel.inc',	 '1000', 'filterlist',  0, 'remove filter',  '', '');	// Remove a filter file
$mnu->Add('filteradd',	'server/filteradd.inc',	 '1000', 'filterlist',  0, 'add new file',  '', 'Create a new empty filter file on server');	// Add a filter file
$mnu->Add('filteredit',	'server/filteredit.inc', '1000', 'filterlist',  0, 'edit file',  '', 'Modify a filter file on server');	// Add a filter file

// 550 - server spam list
$mnu->Add('vxnode551',	'', 					 '1000', '', 	   		0, '', '', '');	//  push submenu down of 1 level (disconnect from root level)
$mnu->Add('spamlist',   'server/spamlist.inc',   '1000', 'vxnode551',   1, 'server local spam list|server spam \t e-mail', '', 'All the emails coming from the listed addresses will be blocked by the server');
$mnu->Add('spamiplist', 'server/spamiplist.inc', '1000', 'vxnode551',   2, 'server local spam list|server spam \t IP address', '', 'Action performed by the server for all the emails coming from each listed IP address');
$mnu->Add('spamadd',	'server/spamadd.inc', 	 '1000', 'spamlist',  	0, 'add new spammer address',  '', 'Add an e-mail address to the local spam address list');	// add spam address
$mnu->Add('spamipadd',	'server/spamipadd.inc',	 '1000', 'spamiplist', 	0, 'add new spammer IP',  '', 'Add an IP address to the local spam address list');	// add spam IP address
$mnu->Add('spamipedit',	'server/spamipadd.inc',	 '1000', 'spamiplist', 	0, 'edit spammer IP',  '', 'Edit an IP address in the local spam address list');	// add spam IP address

// 600 - server logs
$mnu->Add('serverlogs',	'server/serverlogs.inc', '1000', '', 	   		0, 'server logs', '','', '');	// server logs
$mnu->Add('serverlog',	'server/serverlog.inc',  '1000', 'serverlogs', 	0, '$ltype $log', '', '$logtype');	// server log viewing / delete

// 700 - froz messages
$mnu->Add('frozlist',	'server/frozlist.inc',   '1000', '', 	   		0, 'server spool|frozen messages', '', ''); // list frozen msgs
$mnu->Add('frozdel',	'server/frozdel.inc',    '1000', 'frozlist',   	0, 'delete', '', ''); // delete frozen msgs
$mnu->Add('frozsubmit',	'server/frozsubmit.inc', '1000', 'frozlist',   	0, 'resubmit', '', ''); // resubmit frozen msgs
$mnu->Add('frozgetlog',	'server/frozgetlog.inc', '1000', 'frozlist',   	0, 'get log', '', ''); // get frozen msg log
$mnu->Add('frozgetmsg',	'server/frozgetmsg.inc', '1000', 'frozlist',   	0, 'view message', '', ''); // get frozen msg

// 800 - others
$mnu->Add('update',		'update.inc', 			 '1000', '',     		0, 'check for new version', '', ''); // check new version
$mnu->Add('config',		'config.inc', 			 '1000', '',     		0, 'phpxmail config', '', 'PHPXmail configuration options'); // edit configuration
$mnu->Add('legend',		'legend.inc', 			 '1111', '',     		0, 'phpxmail icons legend', '', 'List of all the icons used in this application'); // show icon legend
$mnu->Add('register',	'register.inc', 		 '1111', '',     		0, 'Register', '', 'Create your own new mailbox'); // visitor register

// - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -

// Check if the access is from a regular user, then modify menues accordling
if (!empty($mail_server->xm_ctrl_t)){
	if ($mail_server->xm_ctrl_t == 'r_user'){
		$all = $mail_server->userlist($domain, $username);
		// split in users and lists
		$lists = array();
		foreach ($all as $id => $mlcheck) {
			if (strtoupper($mlcheck[3]) == 'M') {
				$mlname = $username;
				$mnu->Add('mlusrchglst',	'user/mlusrchglst.inc',   '0010', 'userlist',    	3, '$username@$domain \t addresses', '&domain=$domain&mlname=$mlname', 'List of mailing addresses'); // list mailing list users
				$mnu->Add('mlusrchgadd',	'user/mlusrchgadd.inc',   '0010', 'mlusrchglst', 	0, 'add address', '', 'Add a new user address to the mailing list'); // add mailing list users
				$mnu->Add('mlusrchgedit',	'user/mlusrchgedit.inc',  '0010', 'mlusrchglst', 	0, 'edit address', '', 'Edit user rights for this address'); // edit mailing list users
				$mnu->Add('mlusrchgdel',	'user/mlusrchgdel.inc',   '0010', 'mlusrchglst', 	0, 'remove address', '', ''); // del mailing list users
				// remove the Alias menu item
				$mnu->Nodes['aliaslist']['access'] = "1100";
			}
		}
	}
}
// enable/disable the caption (tip) at the top of each page
$mnu->ShowCaption = ($cfg_allow_caption_info == 'on');

// preset the default page to show when access will be denied
$mnu->DeniedPage = 'denied.inc';

// preset special messages (in this case info about the Cookies needs...)
if (empty($_SESSION['InfoAreReadable']) && $action == 'login') {
	// authorized session cannot be initiated and/or mantained, discover why:
	if (ini_get("session.use_cookies")) {
		// session cookies are not enabled on client side
		// in certain systems it happen also if 3rd party cookies are not enabled
		$mnu->SpecialNotes = "</p><img src=gfx/ico_warn.gif> Login will be possible only if your system allows the use of cookies";
	} else {
		// session cookies are not enabled on server side
		// depending by local settings and use, it seems not to be a problem on certain systems when used locally;
		// anyway it's better to report it, so it will not be surprises when used in remote mode
		$mnu->SpecialNotes = "</p><table><tr><td><img src=gfx/ico_warn.gif></td><td> Warning: since in PHP.INI on the server 'session.use_cookies' is not set to true<br> it could prevents the login on certains systems</td></tr></table>";
	}
}

// if the requested action is authorized for this user, then include the relative file
if ($mnu->isValidAction($action, $mail_server->xm_ctrl_t)) {
	include($mnu->FileToInclude);
}

?>
</div>
</body>
</html>